tests/core/test_utilities.py

```diff
@@ -320,6 +320,23 @@ def test_voxelize_volume_no_face_mesh(rectilinear):
         pv.voxelize_volume(pv.PolyData())
 
 
+def test_voxelize_volume_enclosed_bounds():
+    mesh = pv.Sphere()
+    vox = pv.voxelize_volume(mesh, density=0.15, enclosed=True)
+    mesh_bounds = mesh.bounds
+    vox_bounds = vox.bounds
+    for i in range(6):
+        assert vox_bounds[i] * (-1)**i <= mesh_bounds[i] * (-1)**i
+        
+def test_voxelize_enclosed_bounds():
+    mesh = pv.Sphere()
+    vox = pv.voxelize(mesh, density=0.15, enclosed=True)
+    mesh_bounds = mesh.bounds
+    vox_bounds = vox.bounds
+    for i in range(6):
+        assert vox_bounds[i] * (-1)**i <= mesh_bounds[i] * (-1)**i
```

Comment:
Thanks for the suggestion. I've largely implemented what you suggested. A couple of things I couldn't implement:
2. Not being enclosed does not mean the voxelization exceeds all bounds on all axes, it just means they're not guaranteed to fall inside. In practice you get the lower bound matching the input mesh lower bound and the upper bound falling inside.
4. Airplane is unfortunately not manifold so pyvista voxelization doesn't work as it looks for points contained within the mesh.
Also with this test - I realised it isn't a good test for voxelize as depending on the shape of the mesh, the outer cells can be discarded. This is true for the ant example. The test is good for voxelize_volume though and since they both use the same underlying binning function the base grid generated is the same.